article-analyzer/src/main/java/co/inventorsoft/academy/spring/Service/CategoryService.java

```diff
@@ -0,0 +1,45 @@
+package co.inventorsoft.academy.spring.Service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Service
+public class CategoryService {
+    public Set<String> getCategories(List<List<String>> words) {
+
+        Set<String> categories = new HashSet<>();
+        categories.add(mostFrequentWord(words.get(0)));
+        categories.add(mostFrequentWord(words.get(1)));
+        categories.add(mostFrequentWord(words.get(2)));
+        for (List<String> category: words) {
+            categories.add(mostFrequentWord(category));
+        }
+        return categories;
+    }
+
+    private String mostFrequentWord(List<String> words) {
```

Comment:
this algo will not work if in article context are two words which have the same number of occurrences (example article 3 - java and spring)

Reply:
also please commit categories file